### PR TITLE
conv_k7_r2: Disable broken AVX2 code

### PR DESCRIPTION
**Pull request by rear1019**
This has burned too many people. A point release should be considered.

### CODE REVIEW

File: kernels/volk/volk_8u_conv_k7_r2puppet_8u.h [hunk 1]
```diff
@@ -188,87 +188,87 @@ static inline void volk_8u_conv_k7_r2puppet_8u_spiral(unsigned char* syms,
...
-#include <immintrin.h>
-#include <stdio.h>
-
-static inline void volk_8u_conv_k7_r2puppet_8u_avx2(unsigned char* syms,
-                                                    unsigned char* dec,
-                                                    unsigned int framebits)
-{
-
-
-    static int once = 1;
```
Comment (jdemel) [inline kernels/volk/volk_8u_conv_k7_r2puppet_8u.h:202]: Might that be a reason it works in our CI but not in applications?

File: kernels/volk/volk_8u_conv_k7_r2puppet_8u.h [hunk 2]
```diff
@@ -188,87 +188,87 @@ static inline void volk_8u_conv_k7_r2puppet_8u_spiral(unsigned char* syms,
...
-#endif /*LV_HAVE_AVX2*/
+//#if LV_HAVE_AVX2
+//
+//#include <immintrin.h>
+//#include <stdio.h>
+//
+// static inline void volk_8u_conv_k7_r2puppet_8u_avx2(unsigned char* syms,
+//                                                    unsigned char* dec,
+//                                                    unsigned int framebits)
+//{
```
Comment (jdemel) [inline kernels/volk/volk_8u_conv_k7_r2puppet_8u.h:199]: I was really hoping someone would step up to fix this issue. But at this point I doubt it.

**Review by jdemel — APPROVED** [review]
Sad but LGTM.

I'd really like to have this kernel fixed. But unless we find someone with rather intimate knowledge of the algorithm and the code, this might be a really hard problem. Just the length and complexity of the function tells us that we should probably refactor it into multiple functions.

Given that, most FEC algorithms are just too complex to be squashed into a single kernel. They are better off in their distinct module. In order to ensure good performance they need quite a lot of state together with initialization etc.

File: kernels/volk/volk_8u_conv_k7_r2puppet_8u.h (hunk 1 above)
Comment (rear1019) [reply kernels/volk/volk_8u_conv_k7_r2puppet_8u.h:202]: If I remember correctly the puppet function are only used in testing.

**Review by michaelld — APPROVED** [review]
I'm OK with this as-is, though my preference -- mostly being pragmatic -- is to use `#if 0` and `#endif /* 0 */` around the code to be disabled rather than commenting out every single line, because it comes in such easy chunks to do this disabling over. If this were a bunch of smaller pieces of code, then I'd certainly use comments.

**Pull request merged**